src/dynarec/la64/dynarec_la64_emit_math.c

```diff
@@ -171,9 +170,8 @@ void emit_add32c(dynarec_la64_t* dyn, int ninst, rex_t rex, int s1, int64_t c, i
     IFX(X_CF)
     {
         if (rex.w) {
-            MOV32w(x2, 0xffffffff);
```

Comment:
oh, yeah, that was just wrong before. mixing `x2` direct access register with `s1`/`s2` arbitrary scratch register is a sure fail at some point!